### PR TITLE
add wui-input

### PR DESCRIPTION
**Pull request by svenvoskamp**
# Changes

- feat: Add `wui-input` component

### CODE REVIEW

File: packages/ui/src/composites/wui-input/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return html`<input
+        class=${sizeClass}
+        ?disabled=${this.disabled}
+        placeholder=${this.placeholder}
+      />
+      ${this.templateIcon()}`
+  }
+
+  private templateIcon() {
+    const inputSize = this.size === 'md' ? 'lg' : 'md'
```
Comment (0xAsimetriq) [inline packages/ui/src/composites/wui-input/index.ts:34]:
```suggestion
    const iconSize = this.size === 'md' ? 'lg' : 'md'
```

File: packages/ui/src/composites/wui-input/styles.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  @media (hover: hover) and (pointer: fine) {
+    input:hover:enabled {
+      background-color: var(--wui-overlay-010);
+    }
+  }
+
+  wui-icon {
+    position: absolute;
```
Comment (0xAsimetriq) [inline packages/ui/src/composites/wui-input/styles.ts:51]: let's set `pointer-events: none` on the icon, as clicking on it at the moment doesn't focus the input

File: packages/ui/src/composites/wui-input/styles.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  input {
+    width: 100%;
+    border-radius: var(--wui-border-radius-xxs);
+    border: 1px solid var(--wui-overlay-005);
+    background: var(--wui-overlay-005);
+    font-size: var(--wui-font-size-paragraph);
+    font-weight: var(--wui-font-weight-regular);
+    letter-spacing: var(--wui-letter-spacing-paragraph);
+    color: var(--wui-color-fg-100);
+    transition: all 200ms ease-in-out 35ms;
```
Comment (0xAsimetriq) [inline packages/ui/src/composites/wui-input/styles.ts:17]: Can add `carret-color` property to match current accent color for 🔥 points 😄 
Similarly we could also set global `:selection` background color to be accent and text color to be inverse of that.

**Review by 0xAsimetriq — CHANGES_REQUESTED** [review]

File: packages/ui/src/composites/wui-input/styles.ts (hunk 3 above)
Comment (svenvoskamp) [reply packages/ui/src/composites/wui-input/styles.ts:17]: Good one! Will set `caret-color` for now. Let's add `:selection` later on.

**Review by 0xAsimetriq — CHANGES_REQUESTED** [review]
Just noticed that even if icon is not passed, input will have bigger padding-left as if something was still there, can we adjust this to be normal padding when icon is not provided

**Review by 0xAsimetriq — APPROVED** [review]

**Pull request merged**